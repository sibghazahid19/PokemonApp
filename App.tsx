import React from 'react';
import {Provider} from 'react-redux';
import store from './src/store/store';
import MainNavigator from './core/navigation/MainNavigator';

const App = () => (
  <Provider store={store}>
    <MainNavigator testID="main-navigator" />
  </Provider>
);

export default App;
