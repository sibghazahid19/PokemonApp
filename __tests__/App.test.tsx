import React from 'react';
import {render} from '@testing-library/react-native';
import App from '../App';

jest.mock('../src/store/store', () => ({
  getState: jest.fn(),
  subscribe: jest.fn(),
  dispatch: jest.fn(),
}));

jest.mock('../core/navigation/MainNavigator', () => {
  const React = require('react');
  const {Text} = require('react-native');
  return () => <Text testID="MainNavigator">Main Navigator</Text>;
});

describe('App Component', () => {
  it('should render without crashing', () => {
    const {toJSON} = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should provide the Redux store to the application', () => {
    const {getByTestId} = render(<App />);
    expect(getByTestId('MainNavigator')).toBeTruthy();
  });
});
