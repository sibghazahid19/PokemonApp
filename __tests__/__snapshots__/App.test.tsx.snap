// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render without crashing 1`] = `
<Text
  testID="MainNavigator"
>
  Main Navigator
</Text>
`;
